File: server/src/main/scala/rating/models/RatingHistory.scala
package rating.models

import java.sql.Timestamp
import io.circe.generic.JsonCodec, io.circe.syntax._

import TimestampEncoding._

@JsonCodec case class RatingHistory(
  name: String,
  series: List[RatingHistoryRecord]
)

@JsonCodec case class RatingHistoryRecord(
  value: Int,
  name: Timestamp
)

File: client/rating-app-client/src/app/rating-history.ts
import { League } from './league';
import { RatedUser } from './rated-user';
import { WithId } from './with-id';

export class RatingHistory {
  /** league name */
  name: string;
  /** data */
  series: RatingHistoryRecord[];
}

export class RatingHistoryRecord {
  /** rating */
  value: number;
  /** ISO timestamp string. "2016-09-20T01:51:53.101Z" e.g. */
  name: string;
}

